caikit/runtime/model_management/model_manager.py

```diff
@@ -112,10 +112,20 @@ def __init__(self):
 
         # Keep track of whether lazy loading is enabled
         self._lazy_load_local_models = runtime_cfg.lazy_load_local_models
+        error_message = (
+            "runtime.local_models_dir must be a valid path"
+            " if set with runtime.lazy_load_local_models. "
+        )
+        if runtime_cfg.local_models_dir is None:
+            error_message += "runtime.lazy_load_local_models is unset in config file"
+        elif not self._local_models_dir:
+            error_message += f"Provided path: {runtime_cfg.local_models_dir}"
+
```

Comment:
This might be neater if we removed the `value_check` entirely and raised exceptions based on these conditions itself?

Reply:
I can see it making sense having two cases, but since the root is still the same variable value, having it compact like this seems reasonable. 
The combination of the two env vars will raise the error, and the fix is to set the env var. The message error will reflect what type of change to make

Reply:
My only concern with this is that if in the future, if the condition to throw an error changes or another condition needs to be added, then it has to be changed in 2 places, one in the block above and one within the `error.value_check`?

Reply:
```
if self._lazy_load_local_models:
  if runtime_cfg.local_models_dir is None:
     # unset - throw value error with unset
   elif self._local_models_dir is None:
     # set but incorrect - throw value error with invalid path
 ```
or
```
if self._lazy_load_local_models:
  error.value_check(runtime_cfg.local_models_dir is not None)
     # unset - throw value error with unset
   error.value_check(not self._local_models_dir):
     # set but incorrect - throw value error with invalid path
```